projects/core/src/lib/operators/dialogs/column-range-filter/column-range-filter.component.ts

```diff
@@ -72,54 +106,131 @@ export class ColumnRangeFilterComponent implements OnInit, OnDestroy {
                 .subscribe((attributes) => this.attributes$.next(attributes)),
         );
     }
+    selectLayer(layer: Layer): void {
+        if (this.layerDataSubscription) {
+            this.layerDataSubscription.unsubscribe();
+        }
+
+        const dataStream = this.projectService.getLayerDataStream(layer);
+        const metadataStream = this.projectService.getLayerMetadata(layer);
+
+        if (!dataStream || !metadataStream) {
+            return;
+        }
+
+        this.layerDataSubscription = combineLatest([dataStream, metadataStream]).subscribe(
+            ([data, metadata]) => {
+                if (layer instanceof VectorLayer) {
+                    this.processVectorLayer(layer, metadata as VectorLayerMetadata, data);
+                }
+            },
+            (_error) => {
+                // TODO: cope with error
+            },
+        );
+    }
+
+    processVectorLayer(_layer: VectorLayer, metadata: VectorLayerMetadata, data: VectorData): void {
+        if (!data.data || (data.data && data.data.length === 0)) {
+            return;
+        }
+        this.dataStream = data;
+    }
 
     //control over filters
-    get filters(): UntypedFormArray {
-        return this.form.get('filters') as UntypedFormArray;
+    get filters(): FormArray<FormGroup<FilterForm>> {
+        return this.form.get('filters') as FormArray<FormGroup<FilterForm>>;
     }
 
-    newFilter(): UntypedFormGroup {
+    newFilter(): FormGroup<FilterForm> {
         return this.formBuilder.group({
-            attribute: '',
-            ranges: this.formBuilder.array([]),
+            attribute: this.formBuilder.nonNullable.control<string>(''),
+            ranges: this.formBuilder.array<FormGroup<RangeForm>>([]),
         });
     }
 
     addFilter(): void {
         this.filters.push(this.newFilter());
-        this.addRange(this.filters.length - 1);
+        this.addRange(this.filters.length - 1, '', '');
     }
 
     removeFilter(i: number): void {
+        //remove histogram
+        this.removeHistogram(i);
```

Comment:
This is funny with respect to the previous comment about a rather complex function that processes something and has side effects to see this comment :smile: . No action needed here, though :smile: .

Reply:
:-D